### PR TITLE
JAMES-3693 Fixup RedisHealthCheck wrong check when Redis Cluster

### PR DESCRIPTION
**Pull request by vttranlina**
The healthcheck: `org.apache.james.backends.redis.RedisHealthCheck` always log can not connect to Redis.
even when Redis cluster work up normal

## Reason 
- Before invoking method `redisClusterClient.connectAsync` we need to initial partition first by invoking method `redisClusterClient.getPartitions
- The method `redisClusterClient.connectAsync(StringCodec.UTF8)` always success when the cluster fail (eg: 1 /3 node down) => We need one more step for checking. 

## How to fix it
- call `.getPartitions` first, then `.connectAsync` -> `.clusterInfo()`

### CODE REVIEW

File: backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisClusterExtension.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    public RedisClusterExtension(Network network) {
+        this.network = network;
+        redis1.withNetwork(network);
+        redis2.withNetwork(network);
+        redis3.withNetwork(network);
+    }
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) throws IOException, InterruptedException {
+        redis3.start();
```
Comment (quantranhong1999) [inline backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisClusterExtension.java:120]: Depend on would help start redis 1 and 2 also?

File: backends-common/redis/src/main/java/org/apache/james/backends/redis/RedisHealthCheck.scala [hunk 2]
```diff
@@ -21,44 +21,77 @@ package org.apache.james.backends.redis
...
   override def componentName(): ComponentName = redisComponent
 
   override def check(): Publisher[Result] =
-    connectRedis()
-      .`then`(SMono.just(Result.healthy(redisComponent)))
+    healthcheckPerform.check()
       .onErrorResume(_ => SMono.just(Result.degraded(redisComponent, "Can not connect to Redis.")))
+}
+
+trait RedisHealthcheckPerform {
```
Comment (quantranhong1999) [inline backends-common/redis/src/main/java/org/apache/james/backends/redis/RedisHealthCheck.scala:57]: sealed trait IMO

**Review by quantranhong1999 — APPROVED** [review]

File: backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisClusterExtension.java (hunk 1 above)
Comment (vttranlina) [reply backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisClusterExtension.java:120]: yes,

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by hungphan227** [comment]
It seems that RedisHealthCheck has to create new redis connection when check method is run. IMO we should create connection only once

**Comment by vttranlina** [comment]
> It seems that RedisHealthCheck has to create new redis connection when check method is run. IMO we should create connection only once

I based it on the old way, WDYT? @quantranhong1999

**Comment by quantranhong1999** [comment]
> It seems that RedisHealthCheck has to create new redis connection when check method is run. IMO we should create connection only once

If we reuse a single connection, then does Lettuce have a mechanism to recover the connection after a Redis outage? If yes, we can just use a single connection IMO.

**Comment by vttranlina** [comment]
> It seems that RedisHealthCheck has to create new redis connection when check method is run. IMO we should create connection only once

yes, but we create new ones and close immediately after that

**Comment by hungphan227** [comment]
> > It seems that RedisHealthCheck has to create new redis connection when check method is run. IMO we should create connection only once
> 
> yes, but we create new ones and close immediately after that

redisClusterClient is used to create connections. It seems to me that this one is quite heavy and should not be recreated over and over again.

**Comment by vttranlina** [comment]
updated, Use single connection for health check

**Review by chibenwa — APPROVED** [review]

**Review by hungphan227 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**